### PR TITLE
chore(test): simplify workflows

### PR DESCRIPTION
**Pull request by janbrasna**
## Description

Streamlines current Actions for better speed, maintainability & security.

- [x] This PR conforms to the [Contribution Guidelines](/CONTRIBUTING)
- [ ] ~AI tools were used in generating this pull request~

## Issues

Closes: #14

### CODE REVIEW

File: .github/workflows/test.yml
Comment (Trinaa) [inline .github/workflows/test.yml:16]: This seems much better!

**Review by Trinaa — APPROVED** [review]

File: .github/workflows/test.yml
Comment (janbrasna) [reply .github/workflows/test.yml:16]: The diffs don't make much sense after the merging, but I'll make a faux changeset on the side to be able to compare the lines directly, with explainers…

(Also, I need to amend/re-sign, brb.)

Comment (janbrasna) [reply .github/workflows/test.yml:16]: Ah actually you've probably just spotted the main change anyways… Here's the pure diff without any filesystem changes for visibility: https://github.com/janbrasna/ecosystem-test-scripts/commit/a646a8d4bf180c3d1ec821a9efe3301a9ff2b0b7

1. Both .yml files only differ by their trigger, so they're now merged, and the trigger covers both PRs and pushes against _main_.
2. Poetry is installed cleanly into a separate env provided by pipx (that's already by default on the runner) for better separation from the following Python install action (i.e. it doesn't matter what Python version was used as default in the runner, what was used to bootstrap Poetry, and what version then Poetry used to install the project — isolation avoids any clashes, and will make switching versions say for i.e. matrix testing hassle-free).
3. Caching venvs is hard. Basically storing the whole workdir is dangerous wrt secrets, repo tokens, docker api keys etc., anything that's written to the ./~ may be later (re-)used for compromising, so it's better to leave caching and storing artifacts to the machinery available by the default action, which will always be more visible and thoroughly battle tested than boutique scripts trying to achieve the same — unless you have absolutely positive need for own custom logic of course…

Comment (janbrasna) [reply .github/workflows/test.yml:16]: There is also difference in amount of data stored in the caches, as the default tooling knows better what needs picking perhaps, meaning less i/o and less time spent priming and storing it.

Original workflow:
- first run: 45s + 30s priming the cache, subsequent runs: 35s + 25s restoring ~90MB

This PR:
- first run: 25s + 25s priming the cache, subsequent runs: 25s + 15s restoring ~30MB

Comment (Trinaa) [reply .github/workflows/test.yml:16]: Out of curiosity, how granular do people usually go with workflows. Like in this case would separating static analysis (checks) and tests (test-coverage) into two workflows be beneficial?

Comment (janbrasna) [reply .github/workflows/test.yml:16]: I was actually thinking about running `make check && make test*` in one job in succession to save some resources 😇 instead of spinning up two runners for that when none of that testing is destructive. (We often do this, where the ruffs and blacks and linters and formatters are a pre-requisite to anything that follows…)

Most of the time spent is on images, env, binaries etc. — the actual tests take fractions of a second (ok here's mypy which does take some time tho), so for bigger test suites it's often easier to run them sequentially from one environment, than starting many jobs in parallel;) unless you need the isolation.

If you need to run this hundred times a day, I'd actually join the two, with the downside you won't be able to just glance which part failed the whole run without scanning the actual error logs.

If you need this only a few times a week… (:) than glanceability wins I think;)

**Comment by Trinaa** [comment]
@janbrasna I'm wondering if one of us needs to git the 'Merge when ready'? Does git block you?

**Comment by janbrasna** [comment]
@Trinaa Oh I don't have the necessary team/push rights, that's for you to do the honors;)

**Pull request merged**
